vendor/k8s.io/kubernetes/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -472,6 +484,20 @@ func (c *MaxPDVolumeCountChecker) filterVolumes(volumes []v1.Volume, namespace s
 	return nil
 }
 
+// matchProvisioner helps identify if the given PVC belongs to the running predicate.
+func (c *MaxPDVolumeCountChecker) matchProvisioner(pvc *v1.PersistentVolumeClaim) bool {
+	if pvc.Spec.StorageClassName == nil {
+		return false
+	}
+
+	storageClass, err := c.scInfo.GetStorageClassInfo(*pvc.Spec.StorageClassName)
```

Comment:
Is this a significant performance impact in this code path?

Reply:
I ran a few benchmarks when I was working on this (https://github.com/kubernetes/kubernetes/pull/80360), but I didn't noticed any significant performance impact. Unfortunately I didn't save the numbers.